scientific_library/tvb/simulator/simulator.py

```diff
@@ -363,6 +363,38 @@ def _loop_monitor_output(self, step, state):
         if any(outputi is not None for outputi in output):
             return output
 
+    def bound_and_clamp(self, state):
```

Comment:
These operations should be done via code-generation (i.e. numba) to avoid this sort of dynamism at runtime.  The bounding & clamping isn't changing during a simulation.